noir-projects/aztec-nr/aztec/src/macros/utils.nr

```diff
@@ -67,27 +67,48 @@ pub(crate) comptime fn modify_fn_body(body: [Expr], prepend: Quoted, append: Quo
     body_expr.expect(f"Body is not an expression: {body_quote}")
 }
 
-pub(crate) comptime fn add_to_field_slice(slice_name: Quoted, name: Quoted, typ: Type) -> Quoted {
+pub(crate) comptime fn add_to_field_array(
```

Comment:
Can you add a comment here explaining wtf this fn does, ideally with an example invocation?